nwnltr.c

```diff
@@ -128,21 +130,24 @@ void load_ltr(const char *filename, struct ltrfile *ltr) {
 void build_ltr(const char *filename, struct ltrfile *ltr) {
     memset(ltr, 0, sizeof(*ltr));
     strncpy(ltr->header.magic, "LTR V1.0", 8);
-    ltr->header.num_letters = 28;
+    ltr->header.num_letters = MAX_LETTERS;
 
     char buf[256] = {0};
     char *p, *q;
     int count = 0;
     int midcount = 0;
-    while (scanf("%s", buf) == 1) {
+    while (scanf("%255s", buf) == 1) {
         for (q = buf; *q; q++) {
+            if (((*q) == 0x0d) || ((*q) == 0x0a) || ((*q) == '#')) // stop on CR, LF, or #
```

Comment:
How does this ever return LF? Or CR for that matter? `%s` should scan everything that's not whitespace.

Reply:
On mingw32/msys2, it returns a string like "Arthur\n\0", I've never actually seen it return a CR but it definitely returns the LF before the NUL. Why? I don't know.